src/context/client.rs

```diff
@@ -19,48 +20,123 @@ impl RedisModuleClientInfo {
+        } else {
+            Ok(ValkeyString::from_redis_module_string(
+                self.ctx,
+                client_username,
+            ))
+        }
+    }
+
+    /// wrapper for RedisModule_GetClientUserNameById using current client ID
+    pub fn get_client_username(&self) -> ValkeyString {
+        match self.get_client_username_by_id(self.get_client_id()) {
+            Ok(tmp) => tmp,
+            Err(_err) => self.create_string("Client username is null"),
+        }
     }
 
-    pub fn get_client_cert(&self) -> ValkeyString {
+    /// wrapper for RedisModule_GetClientCertificate
+    pub fn get_client_cert(&self) -> ValkeyResult<ValkeyString> {
         let client_id = self.get_client_id();
         let client_cert = unsafe { RedisModule_GetClientCertificate.unwrap()(self.ctx, client_id) };
-        ValkeyString::from_redis_module_string(self.ctx, client_cert)
+        if client_cert.is_null() {
+            Err(ValkeyError::Str("Client cert is null"))
+        } else {
+            Ok(ValkeyString::from_redis_module_string(
+                self.ctx,
+                client_cert,
+            ))
+        }
     }
 
-    pub fn get_client_info(&self) -> RedisModuleClientInfo {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientInfoById
+    pub fn get_client_info_by_id(&self, client_id: u64) -> ValkeyResult<RedisModuleClientInfo> {
         let mut mci = RedisModuleClientInfo::new();
         let mci_ptr: *mut c_void = &mut mci as *mut _ as *mut c_void;
-        unsafe { RedisModule_GetClientInfoById.unwrap()(mci_ptr, client_id) };
-        mci
+        unsafe {
+            RedisModule_GetClientInfoById.unwrap()(mci_ptr, client_id);
+        };
+        if mci_ptr.is_null() {
+            Err(ValkeyError::Str("Client info is null"))
+        } else {
+            Ok(mci)
+        }
+    }
+
+    /// wrapper for RedisModule_GetClientInfoById using current client ID
+    pub fn get_client_info(&self) -> RedisModuleClientInfo {
+        match self.get_client_info_by_id(self.get_client_id()) {
+            Ok(tmp) => tmp,
+            Err(_err) => RedisModuleClientInfo::new(),
+        }
+    }
+
+    /// wrapper to get the client IP address from RedisModuleClientInfo
+    pub fn get_client_ip_by_id(&self, client_id: u64) -> ValkeyResult<String> {
+        let client_info = match self.get_client_info_by_id(client_id) {
+            Ok(tmp) => tmp,
+            Err(_err) => {
+                return Err(ValkeyError::Str("Client info is null"));
+            }
+        };
+        let addr_u8: Vec<u8> = client_info.addr.iter().map(|&x| x as u8).collect();
+        let ip_addr_as_string = String::from_utf8_lossy(&addr_u8)
+            // w/o trim it will be: "127.0.0.1\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00
+            .trim_matches(char::from(0))
+            .to_string();
```

Comment:
An alternative that can be considered:

1. Using Lossy conversion - 

```
let c_str = unsafe { CStr::from_ptr(client_info.addr) };
let string = c_str.to_string_lossy().into_owned();
```

2 - Handling non utf-8 characters -
```
let c_str = unsafe { CStr::from_ptr(client_info.addr) };
match c_str.to_owned().into_string() {
    Ok(x) => {},
    Err(_) => {},
}
```